File: 2766-find-the-prefix-common-array-of-two-arrays/2766-find-the-prefix-common-array-of-two-arrays.java
class Solution {
    public int[] findThePrefixCommonArray(int[] A, int[] B) {
        int[] res = new int[A.length];
        int[] view = new int[A.length + 1];
        int count = 0;
        for (int i = 0; i < A.length; i++) {
            if (view[A[i]]==0) view[A[i]] = 1;
            else if (view[A[i]]==1) count++;
            if (view[B[i]]==0) view[B[i]] = 1;
            else if (view[B[i]]==1) count++;
            res[i] = count;
        }
        return res;
    }
}

File: 2766-find-the-prefix-common-array-of-two-arrays/2766-find-the-prefix-common-array-of-two-arrays.py
class Solution:
    def findThePrefixCommonArray(self, A: List[int], B: List[int]) -> List[int]:
        da,db = {},{}
        for i in range(len(A)):
            if A[i] not in da:
                da[A[i]] = i
            if B[i] not in db:
                db[B[i]] = i
        res = [0]*len(A)
        for i in range(len(A)):
            count = 0
            for j in range(i+1):
                if da[A[j]] <= i and db[A[j]] <= i:
                    count += 1
            res[i] = count
        return res